main.appinsights.tf

```diff
@@ -4,7 +4,7 @@ resource "azurerm_application_insights" "this" {
   application_type    = "web"
   location            = var.location
   name                = "app-insights-${var.name}"
-  resource_group_name = var.resource_group.name
-  tags                = var.tags
+  resource_group_name = var.resource_group_name
+  tags                = var.application_insights.tags == null ? var.tags : var.application_insights.tags == {} ? {} : var.application_insights.tags
```

Comment:
Should we not merge both tags in case they are specified in both places ? I would just remove tags from the application insights object and jsut use the main tags.